src/sentry/integrations/utils/metrics.py

```diff
@@ -329,6 +334,25 @@ def __exit__(
             )
 
 
+class IntegrationEventLifecycle(EventLifecycle):
```

Comment:
the logic makes sense but I think I;m missing some context?  Why make a new `EventLifecycle` class? Is this a new pattern/convention we're going for where instead of halting at the callsite, it's a halt at the context manager level?